File: Day211_2025/Q211_GFG.py
class Solution:
    def cntSubarrays(self, arr, k):
        prefix_sums = {}
        res = 0
        curr_sum = 0

        for num in arr:
            curr_sum += num

            if curr_sum == k:
                res += 1

            if (curr_sum - k) in prefix_sums:
                res += prefix_sums[curr_sum - k]

            prefix_sums[curr_sum] = prefix_sums.get(curr_sum, 0) + 1

        return res

File: Day211_2025/Q211_GFG.java
import java.util.*;

class Q211_GFG {
    public int cntSubarrays(int[] arr, int k) {
        Map<Integer, Integer> prefixSums = new HashMap<>();
        int res = 0;
        int currSum = 0;

        for (int i = 0; i < arr.length; i++) {
            currSum += arr[i];

            if (currSum == k)
                res++;

            if (prefixSums.containsKey(currSum - k))
                res += prefixSums.get(currSum - k);

            prefixSums.put(currSum, prefixSums.getOrDefault(currSum, 0) + 1);
        }

        return res;
    }
}
